warp10/src/main/java/io/warp10/script/ext/urlfetch/UrlFetchWarpScriptExtension.java

```diff
@@ -0,0 +1,146 @@
+  public static final String ATTRIBUTE_URLFETCH_MAXSIZE_HARD = "urlfetch.maxsize.hard";
+
+  /**
+   * Current  URLFETCH so far in the sessions
+   */
+  public static final String ATTRIBUTE_URLFETCH_SIZE = "urlfetch.size";
+
+  //
+  // DEFAULTS
+  //
+
+  public static final long DEFAULT_URLFETCH_LIMIT = 64L;
+  public static final long DEFAULT_URLFETCH_MAXSIZE = 1000000L;
+
+  //
+  // ASSOCIATIONS attributes to either configuration or defaults
+  //
+
+  /**
+   * Associates the attribute name to the configuration name
+   */
+  private static final Map<String, String> attributeToConf;
+
+  /**
+   * Associates the attribute name to its default value
+   */
+  private static final Map<String, Long> attributeToDefault;
+
+  private static final Map<String, Object> functions;
+
+  static {
+    // Initialize attribute->configuration
+    Map<String, String> a2c = new HashMap<String, String>();
+    a2c.put(ATTRIBUTE_URLFETCH_LIMIT, WARPSCRIPT_URLFETCH_LIMIT);
+    a2c.put(ATTRIBUTE_URLFETCH_MAXSIZE, WARPSCRIPT_URLFETCH_MAXSIZE);
+    a2c.put(ATTRIBUTE_URLFETCH_LIMIT_HARD, WARPSCRIPT_URLFETCH_LIMIT_HARD);
+    a2c.put(ATTRIBUTE_URLFETCH_MAXSIZE_HARD, WARPSCRIPT_URLFETCH_MAXSIZE_HARD);
+    attributeToConf = Collections.unmodifiableMap(a2c);
+
+    // Initialize attribute->default
+    Map<String, Long> a2d = new HashMap<String, Long>();
+    a2d.put(ATTRIBUTE_URLFETCH_LIMIT, DEFAULT_URLFETCH_LIMIT);
+    a2d.put(ATTRIBUTE_URLFETCH_MAXSIZE, DEFAULT_URLFETCH_MAXSIZE);
+    a2d.put(ATTRIBUTE_URLFETCH_LIMIT_HARD, DEFAULT_URLFETCH_LIMIT);
+    a2d.put(ATTRIBUTE_URLFETCH_MAXSIZE_HARD, DEFAULT_URLFETCH_MAXSIZE);
+    attributeToDefault = Collections.unmodifiableMap(a2d);
+
+    // Create functions and map
+    functions = new HashMap<String, Object>();
+
+    functions.put("URLFETCH", new URLFETCH("URLFETCH"));
+    functions.put("MAXURLFETCHCOUNT", new MAXURLFETCHCOUNT("MAXURLFETCHCOUNT"));
+    functions.put("MAXURLFETCHSIZE", new MAXURLFETCHSIZE("MAXURLFETCHSIZE"));
+  }
+
+  public Map<String, Object> getFunctions() {
+    return functions;
+  }
+
+  /**
+   * Get the value of the attribute in the stack, if not present get the value in the configuration and it not present either, get the default value.
+   * Also set the attribute in the stack once the value is found.
+   * @param stack The stack the get the attribute from, if present.
+   * @param attribute The attribute name to get.
+   * @return The first available value in the list: attribute value, configuration value, default.
+   */
+  public static Long getAttribute(WarpScriptStack stack, String attribute) {
+    Object ufLimit = stack.getAttribute(attribute);
+
+    if (null != ufLimit) {
+      return (Long) ufLimit;
+    }
+
+    String associatedConf = attributeToConf.get(attribute);
+    long associatedDefault = attributeToDefault.get(attribute);
+    String ufLimitConf = WarpConfig.getProperties().getProperty(associatedConf, Long.toString(associatedDefault));
```

Comment:
The call to WarpConfig.getProperties is costly, you should do it only once at extension registration time.